armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -117,8 +152,7 @@ class AxialLink(typing.Generic[Comp]):
     * :attr:`AxialAssemblyLinkage.linkedComponents`
     """
 
-    lower: typing.Optional[Comp] = dataclasses.field(default=None)
-    upper: typing.Optional[Comp] = dataclasses.field(default=None)
```

Comment:
So turns out, we never use `upper` and having it complicated the new linking. We can bring it back if you feel strongly.